cli/tests/services.bats

```diff
@@ -153,6 +160,114 @@ EOF
+  setup_start_counter_services
+
+  run "$FLOX_BIN" activate --start-services -- bash <(cat <<'EOF'
+    source "${TESTS_DIR}/services/register_cleanup.sh"
+    "$FLOX_BIN" services restart one two invalid
+EOF
+)
+  assert_failure
+  assert_output --partial "❌ ERROR: Service 'invalid' not found"
+
+  run cat start_counter.one
+  assert_output "1"
+  run cat start_counter.two
+  assert_output "1"
+  run cat start_counter.sleeping
+  assert_output "1"
+}
+
+# bats test_tags=services:restart
+@test "restart: restarts a single service" {
+  export FLOX_FEATURES_SERVICES=true
+  setup_start_counter_services
+
+  run "$FLOX_BIN" activate --start-services -- bash <(cat <<'EOF'
+    source "${TESTS_DIR}/services/register_cleanup.sh"
+    "$FLOX_BIN" services restart one
+EOF
+)
+  assert_success
+  assert_output --partial "✅ Service 'one' restarted"
+
+  run cat start_counter.one
+  assert_output "2"
+  run cat start_counter.two
+  assert_output "1"
+  run cat start_counter.sleeping
+  assert_output "1"
+}
+
+# bats test_tags=services:restart
+@test "restart: restarts multiple services" {
+  export FLOX_FEATURES_SERVICES=true
+  setup_start_counter_services
+
+  run "$FLOX_BIN" activate --start-services -- bash <(cat <<'EOF'
+    source "${TESTS_DIR}/services/register_cleanup.sh"
+    "$FLOX_BIN" services restart one two
+EOF
+)
+  assert_success
+  assert_output --partial "✅ Service 'one' restarted"
+  assert_output --partial "✅ Service 'two' restarted"
+
+  run cat start_counter.one
+  assert_output "2"
+  run cat start_counter.two
+  assert_output "2"
+  run cat start_counter.sleeping
+  assert_output "1"
+}
+
+# bats test_tags=services:restart
+@test "restart: restarts all services (incl. running and completed)" {
+  export FLOX_FEATURES_SERVICES=true
+  setup_start_counter_services
+
+  run "$FLOX_BIN" activate --start-services -- bash <(cat <<'EOF'
+    source "${TESTS_DIR}/services/register_cleanup.sh"
+    "$FLOX_BIN" services restart
+EOF
+)
+  assert_success
+  assert_output --partial "✅ Service 'one' restarted"
+  assert_output --partial "✅ Service 'two' restarted"
+  assert_output --partial "✅ Service 'sleeping' restarted"
+
+  run cat start_counter.one
+  assert_output "2"
+  run cat start_counter.two
+  assert_output "2"
+  run cat start_counter.sleeping
+  assert_output "2"
+}
+
+# bats test_tags=services:restart
+@test "restart: restarts stopped services" {
+  export FLOX_FEATURES_SERVICES=true
+  setup_start_counter_services
+
+  run "$FLOX_BIN" activate --start-services -- bash <(cat <<'EOF'
+    source "${TESTS_DIR}/services/register_cleanup.sh"
+    "$FLOX_BIN" services stop sleeping
+    "$FLOX_BIN" services restart sleeping
+EOF
+)
+  assert_success
+  assert_output --partial "✅ Service 'sleeping' stopped"
+  assert_output --partial "✅ Service 'sleeping' restarted"
+
+  run cat start_counter.sleeping
+  assert_output "2"
+}
+
+# ---------------------------------------------------------------------------- #
+
```

Comment:
nonblocking question: I'm curious to hear why you opted for integration tests. I used unit tests for everything interacting with a pre-existing `process-compose` for the `start` ticket

Reply:
It looks like you have both in https://github.com/flox/flox/pull/1913?

Yannik and I talked a bit about testing approaches in [this thread](https://flox-dev.slack.com/archives/C05P6A5J6U8/p1723112691776299?thread_ts=1723110899.013709&cid=C05P6A5J6U8) yesterday. I think the tl;dr is that we're still figuring out what the right balance is.

I took a mostly unit testing approach to `flox services status` because there was a bunch of internal logic to cover and having structured data also made it easier to inject/inspect the results. However I wasn't left feeling super happy that we have enough end-to-end coverage of how the individual parts plumb together, e.g. what user interactions do we support and how does `process-compose` behave in reality.

So for this one I started from the outside, whilst the internal logic is very light, and I'll see what it looks like when I get to the ephemeral activation changes.

Reply:
> It looks like you have both in #1913?

I used integration for anything starting a new process-compose, and I used unit for anything interacting with an existing one

> Yannik and I talked a bit about testing approaches in [this thread](https://flox-dev.slack.com/archives/C05P6A5J6U8/p1723112691776299?thread_ts=1723110899.013709&cid=C05P6A5J6U8) yesterday. I think the tl;dr is that we're still figuring out what the right balance is.
> 
> I took a mostly unit testing approach to `flox services status` because there was a bunch of internal logic to cover and having structured data also made it easier to inject/inspect the results. However I wasn't left feeling super happy that we have enough end-to-end coverage of how the individual parts plumb together, e.g. what user interactions do we support and how does `process-compose` behave in reality.
> 
> So for this one I started from the outside, whilst the internal logic is very light, and I'll see what it looks like when I get to the ephemeral activation changes.

👍 nothing actionable, I was just curious to hear your thoughts